### PR TITLE
path fixing when copying folders from windows

### PR DESCRIPTION
**Pull request by moscoquera**
as window's file separator is '\', when copying folders from windows the destination path get malformed, replacing '\' with '/' is enough to solve it.

### CODE REVIEW

**Comment by pkittenis** [comment]
Hi there, thanks for the PR.

If I understand correctly, `os.path.sep` is `\` on Windows but SFTP file paths always use `/`, therefore file copy with directory paths breaks on Windows?

In that case it would be more appropriate to not use `os.path` in the first place and always use `/` instead as that's what the protocol uses.

**Comment by moscoquera** [comment]
Hi.

imagine a windows folder called parent and with 2 files inside called childrenA.txt and childrenB.txt:

* parent:
       childrenA.txt
       childrenB.txt

to access them python will use "parent\childrenA.txt" and "parent\childrenB.txt", if you are trying to copy parent to a remote sever, you will get 3 files:

- a folder called "parent".
- an two files called "parent\childrenA.txt" and "parent\childrenB.txt"

when the expected behavior will be a single folder called "parent" with 2 files inside.

**Comment by pkittenis** [comment]
Yes, I understand. 

The change needed here is for `remote_path` (only remote path) to always use `/` for joining. Ie `remote_path = '/'.join([remote_dir, file_name])` in both `copy_file` and `copy_remote_file`.

Replacing `os.path`, which is `\` on Windows, with `/` after the path is generated does not make much sense. So if you'd like to make that change, I'd be happy to review it.

**Comment by moscoquera** [comment]
Hi again. sorry for the very late response, i have been busy upgrading a project. i just did the requested changes.

i'd be happy if u can review it.

File: pssh/ssh_client.py [hunk 1]
```diff
@@ -364,7 +364,7 @@ def _copy_dir(self, local_dir, remote_dir, sftp):
         file_list = os.listdir(local_dir)
         for file_name in file_list:
             local_path = os.path.join(local_dir, file_name)
-            remote_path = os.path.join(remote_dir, file_name)
+            remote_path = remote_dir+'/'+file_name
```
[the remark this answers is not in the document]
Comment (moscoquera) [reply pssh/ssh_client.py]: but, isn't it creating a new array each time?

**Comment by moscoquera** [comment]
changes are made

File: pssh/ssh_client.py [hunk 2]
```diff
@@ -364,7 +364,7 @@ def _copy_dir(self, local_dir, remote_dir, sftp):
         file_list = os.listdir(local_dir)
         for file_name in file_list:
             local_path = os.path.join(local_dir, file_name)
-            remote_path = os.path.join(remote_dir, file_name)
+            remote_path = '/'.join([remote_dir,file_name])
```
Comment (pkittenis) [inline pssh/ssh_client.py]: Flake errors: `pssh/ssh_client.py:367:47: E231 missing whitespace after ','`

Per https://travis-ci.org/ParallelSSH/parallel-ssh/jobs/279101695

File: pssh/ssh_client.py [hunk 3]
```diff
@@ -454,8 +454,8 @@ def copy_remote_file(self, remote_file, local_file, recurse=False,
 
     def _copy_remote_dir(self, file_list, remote_dir, local_dir, sftp):
         for file_name in file_list:
-            remote_path = os.path.join(remote_dir, file_name)
-            local_path = os.path.join(local_dir, file_name)
+            remote_path = '/'.join([remote_dir, file_name])
+            local_path = os.path.join(local_dir,file_name)
```
Comment (pkittenis) [inline pssh/ssh_client.py]: Ditto - https://travis-ci.org/ParallelSSH/parallel-ssh/jobs/279101695

**Comment by pkittenis** [comment]
Thanks, changes look good. Only couple minor changes needed for the flake check to pass, see in line comments and [CI job](https://travis-ci.org/ParallelSSH/parallel-ssh/jobs/279101695), looks ready otherwise.

Thanks again for the PR.

**Comment by moscoquera** [comment]
sure, no problem.

**Comment by pkittenis** [comment]
Thanks for making these changes :)

FYI, have found a couple more places where remote path needs changing for windows which are used when `recurse` is enabled. Those changes, along with this fix, will be in next release.

**Pull request merged**